pkg/tarfs/tarfs.go

```diff
@@ -457,18 +457,38 @@ func (f *FS) ReadFile(name string) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	if i.h.FileInfo().Mode().Type()&fs.ModeSymlink != 0 {
+
+	dataSize := i.h.Size
+	typ := i.h.FileInfo().Mode().Type()
+	var r *tar.Reader
+	switch {
+	case typ.IsRegular() && i.h.Typeflag != tar.TypeLink:
+		r = tar.NewReader(io.NewSectionReader(f.r, i.off, i.sz))
+	case typ.IsRegular() && i.h.Typeflag == tar.TypeLink:
```

Comment:
curious would all this just be equivalent to `return f.ReadFile(i.h.Linkname)`? If so, should we just combine this case and the symlink check into a single case?